src/pages/[lang]/projects/[project]/request.astro

```diff
@@ -0,0 +1,39 @@
+---
+import BaseLayout from '@layouts/BaseLayout.astro';
+import { createSupabaseServerClient } from '@backend/supabaseServerClient';
+import { getMyProfile } from '@backend/crud';
+import { getLangFromUrl, getTranslations } from '@i18n';
+import { ProjectRequest } from '@apps/project-request';
+import { getProjectExtended } from '@backend/helpers';
+
+const lang = getLangFromUrl(Astro.url);
+
+const i18n = getTranslations(Astro.request, 'project-request');
+
+const supabase = await createSupabaseServerClient(Astro.cookies);
+
+const projectId = Astro.params.project;
+
+const project = await getProjectExtended(supabase, projectId as string);
+
+const profile = await getMyProfile(supabase);
+if (profile.error || !profile.data) {
+  return Astro.redirect(`/${lang}/sign-in?redirect-to=${Astro.url}`);
+}
+---
+
+<BaseLayout title='join'>
+  <ProjectRequest
+    client:only='react'
+    i18n={i18n}
+    projectId={projectId as string}
+    user={profile.data}
+    isAlreadyMember={project.data}
+  />
+</BaseLayout>
+
+<style>
+  h1 {
```

Comment:
I don't think this will do anything, since the Astro styles are scoped. If you want the h1 to apply on the page globally, you'd need to use either

```
<style is:global>
  h1 { ... }
</style>
```

or 

```
<style>
  :global(h1) { ... }
</style>
```

or (better IMO) move it into the stylesheet of the React component.

Reply:
Agreed. Honestly I need to go back a clean up some things before merging. I will update the PR.